### PR TITLE
Refactor lists into global scss

### PR DESCRIPTION
**Pull request by joelanman**
- move the govuk list styles (normal, bullet and number) to `globals/scss/_lists`
- remove the list component - update all components that use the list component
- remove all code and references to the steps variant - if it's MVP we can add again in future but it doesn't seem part of core styles

This work allows easy styling of lists without having to use the component. This is useful for example when we do a styled wrapper like govspeak.

### CODE REVIEW

**Comment by 36degrees** [comment]
I think keeping the list component is just going to muddy things and cause confusion - I think we probably need to refactor the error summary component to generate the list without using the list macro?

If we're going to move the steps to a new component I think that needs to be done at the same time.

**Comment by joelanman** [comment]
Is steps really important to MVP? I would suggest not?

**Comment by joelanman** [comment]
I've now removed the list component completely and updated the components that used it.

**Comment by 36degrees** [comment]
There's a few things within the packages folder that will need updating:

- The error summary component will need the dependency on the list component removing from its package.json
- The all package will need the dependency on the list component removing from its package.json

I'm unclear as to whether that should happen as part of this PR or when we publish the next release – @igloosi @alex-ju do you know what the right approach would be?

File: src/views/examples/typography/index.njk [hunk 1]
```diff
@@ -111,7 +111,21 @@
       <p class="govuk-body-l govuk-!-w-bold">govuk-body-l + govuk-!-w-bold</p>
 
     </section>
-
+    <section class="govuk-!-mt-r8">
+      <h2 class="govuk-heading-l govuk-!-pb-r2" style="border-bottom: 4px solid;">Lists</h2>
```
Comment (36degrees) [inline src/views/examples/typography/index.njk]: Should we have an example of a list with neither bullets nor numbers?

Comment (joelanman) [reply src/views/examples/typography/index.njk]: I think only if we actually do that on gov.uk anywhere - I'll have a look around

Comment (36degrees) [reply src/views/examples/typography/index.njk]: Lists of links in the related widget tend not to use bullets.

Comment (joelanman) [reply src/views/examples/typography/index.njk]: thanks - adding now

**Comment by joelanman** [comment]
I've added an example for a normal list.

I've removed references to the list component from packages.

I think we should do steps as another PR if needed, otherwise it's blocking the work on the prose scope.

**Comment by 36degrees** [comment]
@joelanman in rebasing this I think we lost the change of the list classes for the component and example lists, which were combined to create one page in #387. It's currently rendering incorrectly on http://govuk-frontend-review.herokuapp.com/.

**Comment by joelanman** [comment]
sorry - will fix now

**Pull request merged**